common/rpc/encryption/localStoreCertProvider.go

```diff
@@ -456,3 +509,64 @@ func appendError(aggregatedErr error, err error) error {
 	}
 	return fmt.Errorf("%v, %w", aggregatedErr, err)
 }
+
+func (s *localStoreCertProvider) refreshCerts() {
+
+	for {
+		select {
+		case <-s.stop:
+			break
+		case <-s.ticker.C:
+		}
+
+		newCerts, err := s.loadCerts()
+		if err != nil {
+			s.logger.Error("failed to load certificates", tag.Error(err))
+			continue
+		}
+
+		if s.certs.isEqual(newCerts) {
+			continue
+		}
+
+		s.RWMutex.Lock()
+		s.certs = newCerts
```

Comment:
can we log if we detected a change in certificates?

also, given that we pretty much already did the hard work of reading and loading the certificates, I'm wondering if the complexity of the isEqual function is worth saving the lock/unlock

Reply:
That was the whole reason to have isEqual - to avoid spamming on every refresh with no change. But I forgot to add the logger.Info(). Added now.